app/assets/javascripts/discourse/app/components/sidebar/filter.gjs

```diff
@@ -23,6 +28,15 @@ export default class Filter extends Component {
   @bind
   teardown() {
     this.sidebarState.clearFilter();
+    document.documentElement.removeEventListener("keydown", this.handleKeydown);
+  }
+
+  @bind
+  handleKeydown(event) {
+    if (event.key === "Escape" && this.sidebarState.filter.length > 0) {
```

Comment:
We should also do the second part of Joffrey's request:

> escape when input has no content: blur

It's fine to do it in the same PR.